### PR TITLE
Update requirements.txt

### PR DESCRIPTION
**Pull request by Antauri**
```
  boto3==1.18.12 (from dbt-athena==1.0.1->-r requirements.in (line 14))
  boto3<2.0.0,>=1.20.17 (from awswrangler==2.14.0->-r requirements.in (line 13))
```
Maybe allowing it to work with awswrangler.

### CODE REVIEW

**Comment by jankatins** [comment]
I think you have to update the version constraints in setup.py: these (as far as I know) will be the relevant ones when installing it as a dependency.

File: setup.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def _dbt_athena_version() -> str:
     install_requires=[
         "dbt-core~=1.0.0",
```
Comment (jankatins) [inline setup.py:55]: This should probably also be adjusted to `>=`or `~=1.0`

I would actually change all of these to `>=` and leave the responsibility to pick a working set o f packages to the end user. Given that the current set by the pinned versions also fails in some scenarios, it at least gives the user the flexibility to use something which works.

Comment (Antauri) [reply setup.py]: Updated. Is it ok?

**Comment by jankatins** [comment]
My understanding of dependencies in `requirements.txt` and `setup.py` is the following:

* `setup.py` dependencies is used when the software is installed as a dependency via pip (e.g. someone runs `pip install dbt-athena`). To prevent version clashes with other packages co-installed with `dbt-athena`, in `setup.py` dependencies, only known not-working versions should be excluded (e.g. it only works after a feature got introduced -> use `>=<version_which_got_feature`). The user is responsible for setting up a consistent working set of packages.
* `requirements.txt` is usually used to specify the dependency of an app which gets deployed (similar to a lock file in other languages). installing it, it should result in a consistent and working set of packages. Packages should therefore be pinned to not get a surprising upgrade when installed later. (I also use `requirement.txt(.in)` sometimes in a laxer way, but I always deploy from a pinned set of packages, e.g. I generate it via  `pip freeze` which has all packages pinned to a specific version).

Going by that and seeing that `requirements.txt` is used in the `Makefile`(so more in a deployment mode), I would currently leave it up to @Tomme to decide if the `requirements.txt` should be adjusted or not and only touch `setup.py` for now.

**Comment by ConstantinoSchillebeeckx** [comment]
@Tomme can you take a look - I suspect this is going to be blocking for most people using this library along with libraries like `awswrangler`

**Pull request merged**